### PR TITLE
Ec-subject form

### LINKED ISSUE #16: Make a form for UCSBSubjects to allow data entry
Make a form for UCSBSubjects to allow data entry

Model your form after UCSBDateForm

- [x] There is a form component under `frontend/src/main/components/UCSBSubjects/UCSBSubjectForm
- [x]  That component has appropriate fields that match those in the database table
- [x] There is data validation appropriate to the various fields
- [x] There is a storybook entry that shows how the form works.
- [x] There is 100% test coverage from npm run coverage
- [x] There is 100% test coverage from npm run coverage and npx stryker run

### PR DESCRIPTION
**Pull request by edisonnchen**
# Overview

In this PR, we implement the form for creating a subject and its tests.

# Issues Addressed

Close #16

### CODE REVIEW

**Comment by edisonnchen** [comment]
Windows might've auto-changed some setting on package-lock

**Comment by edisonnchen** [comment]
Package-lock fix has been pushed

File: frontend/src/tests/components/UCSBSubjectsTest/UCSBSubject.test.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import { render, waitFor, fireEvent } from '@testing-library/react';
```
Comment (bobhe9606) [inline frontend/src/tests/components/UCSBSubjectsTest/UCSBSubject.test.js:1]: I tried to be consistent with UCSBDates so I named the folder and file to be:

UCSBSubjects
UCSBSubjectsForm.test.js

File: frontend/src/main/components/UCSBSubjects/UCSBSubjectForm.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          isInvalid={Boolean(errors.relatedDeptCode)}
+          {...register('relatedDeptCode', {
+            required: 'relatedDeptCode is required.',
+          })}
+        />
+        <Form.Control.Feedback type="invalid">
+          {errors.relatedDeptCode?.message}
+        </Form.Control.Feedback>
+      </Form.Group>
+
```
Comment (bobhe9606) [inline frontend/src/main/components/UCSBSubjects/UCSBSubjectForm.js:130]: Another slot for inactive?

**Comment by bobhe9606** [comment]
LGTM

**Review by AHuangHe — APPROVED** [review]
LGTM

**Pull request merged**